src/esmock.d.ts

```diff
@@ -1,3 +1,20 @@
+type esmockArgs = [
```

Comment:
If this shorthand is giving you trouble, I would highly recommend just repeating things in your d.ts file. When you're writing it by hand, you're forced to do that sometimes. If you were in typescript from the start, this would be simpler, of course.

Making these aliases will just make everyone's editor experience worse, as we won't show any parameters in the tooltips, or their docs.

Reply:
things are reverted back to using the longer typed function signatures

the only problem is I'm not able to figure out how to type this `esmock.partial.p` while at the same time declaring `esmock.partial` as a function

Reply:
the function signatures are repeated now and the typescript file is getting big. marking as resolved.